backend/src/services/secret-v2-bridge/secret-v2-bridge-service.ts

```diff
@@ -686,27 +770,44 @@ export const secretV2BridgeServiceFactory = ({
       projectId
     });
 
-    const decryptedSecrets = secrets.map((secret) =>
-      reshapeBridgeSecret(projectId, environment, groupedPaths[secret.folderId][0].path, {
-        ...secret,
-        value: secret.encryptedValue
-          ? secretManagerDecryptor({ cipherTextBlob: secret.encryptedValue }).toString()
-          : "",
-        comment: secret.encryptedComment
-          ? secretManagerDecryptor({ cipherTextBlob: secret.encryptedComment }).toString()
-          : ""
-      })
-    );
+    const decryptedSecrets = secrets
+      .filter((el) =>
+        permission.can(
+          ProjectPermissionActions.Read,
+          subject(ProjectPermissionSub.Secrets, {
+            environment,
+            secretPath: groupedPaths[el.folderId][0].path,
+            secretName: el.key,
+            secretTags: el.tags.map((i) => i.slug)
+          })
+        )
+      )
```

Comment:
this too will misalign secrets with secrets count when doing pagination right?

Reply:
Addressed already. The mismatch is considered oke